backend/ticketvise/views/api/ticket.py

```diff
@@ -606,9 +592,123 @@ def get_object(self):
 
 class TicketTitleAPIView(UpdateAPIView):
     permission_classes = [UserIsTicketAuthorOrInboxStaffPermission]
-    serializer_class = TicketTitleUpdateSerializer
+
+    def get_serializer(self, *args, **kwargs):
+        return TicketSerializer(fields=["title"], *args, **kwargs)
+
+    def get_object(self):
+        inbox = get_object_or_404(Inbox, pk=self.kwargs["inbox_id"])
+
+        return Ticket.objects.get(inbox=inbox, ticket_inbox_id=self.kwargs["ticket_inbox_id"])
+
+
+class TicketPublishAPIView(UpdateAPIView):
+    permission_classes = [UserIsTicketAuthorOrInboxStaffPermission]
+
+    def get_serializer(self, *args, **kwargs):
+        return TicketSerializer(fields=["is_public", "is_anonymous"], *args, **kwargs)
+
+    def get_object(self):
+        inbox = get_object_or_404(Inbox, pk=self.kwargs["inbox_id"])
+
+        return Ticket.objects.get(inbox=inbox, ticket_inbox_id=self.kwargs["ticket_inbox_id"])
+
+    def update(self, request, *args, **kwargs):
+        # Tickets can only be made public, not hidden again.
+        ticket = self.get_object()
+        ticket.is_public = timezone.now()
```

Comment:
Is het hier niet handig om eerst nog te checken of de ticket niet al public is? Anders wordt de public datum verniewd.